### PR TITLE
!up command to show status of Netsoc websites

### PR DESCRIPTION
**Pull request by OisinA**
Attempts to connect to various Netsoc websites and displays the result.

image

### CODE REVIEW

File: commands/upCommand.go [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+func checkStatus(site string, retryCount int, statuses chan statusCheck) {
+	var timeTaken int64 = 0
+
+	err := try.Do(func(attempt int) (bool, error) {
+		startTime := time.Now()
+		resp, err := http.Get(site)
+		timeTaken = time.Now().Sub(startTime).Milliseconds()
+
+		if resp != nil {
+			if resp.StatusCode != 200 {
```
Comment (ericm) [inline commands/upCommand.go:75]: Looks good. What if the website returns a 307/308 for a www to non www redirect for example. Will it be registered as down?

File: commands/upCommand.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+func checkUpCommand(ctx context.Context, s *discordgo.Session, m *discordgo.MessageCreate) {
+	// Check if committee channel, don't allow in public server
+	if isCommittee(m) {
+		return
+	}
+
+	sites := strings.Split(viper.GetString("netsoc.sites"), ",")
+
+	// Run on a separate goroutine to not block bot
+	go checkStatuses(s, m, sites)
```
[the remark this answers is not in the document]
Comment (OisinA) [reply commands/upCommand.go:35]: Huh wasn't aware, cool 👍

File: commands/upCommand.go (hunk 1 above)
Comment (OisinA) [reply commands/upCommand.go:75]: Hmm better to check against a list of possible error codes instead?

Comment (ericm) [reply commands/upCommand.go:75]: Yeah I think so. Only ones i can think of off the top of my head are 200, 307 and 308

Comment (ocanty) [reply commands/upCommand.go:75]: >= 400 will prolly work

Comment (ericm) [reply commands/upCommand.go:75]: Yeah I think so

**Comment by Strum355** [comment]
I would probably do requests with context timeout, else you could end up hanging requests indefinitely

**Pull request merged**